pkg/build/api/validation/validation.go

```diff
@@ -225,18 +226,39 @@ func validateGitSource(git *buildapi.GitBuildSource) fielderrors.ValidationError
 	return allErrs
 }
 
-func validateImageSource(imageSource *buildapi.ImageSource) fielderrors.ValidationErrorList {
+func validateImageSources(imageSources []buildapi.ImageSource) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
-	allErrs = append(allErrs, validateFromImageReference(&imageSource.From).Prefix("from")...)
-	if imageSource.PullSecret != nil {
-		allErrs = append(allErrs, validateSecretRef(imageSource.PullSecret).Prefix("pullSecret")...)
-	}
-	if len(imageSource.Paths) == 0 {
-		allErrs = append(allErrs, fielderrors.NewFieldRequired("paths"))
+	for _, imageSource := range imageSources {
+		allErrs = append(allErrs, validateFromImageReference(&imageSource.From).Prefix("from")...)
+		if imageSource.PullSecret != nil {
+			allErrs = append(allErrs, validateSecretRef(imageSource.PullSecret).Prefix("pullSecret")...)
+		}
+		if len(imageSource.Paths) == 0 {
+			allErrs = append(allErrs, fielderrors.NewFieldRequired("paths"))
+		}
+		allErrs = append(allErrs, validateImageSourcePaths(imageSource.Paths).Prefix("paths")...)
```

Comment:
Alternatively, you can iterate over the list in the calling function and leave this method as is.